pkg/devices/lock.go

```diff
@@ -45,6 +45,14 @@ func lockRead(device *sdk.Device) ([]*sdk.Reading, error) {
 		lockState = stateLock
 	}
 
+	dState, ok := deviceState[device.ID()]
+
+	if ok {
+		if _, ok := dState[lockState]; ok {
+			lockState = dState[lockState].(string)
```

Comment:
We'll need to remove the global lock state here, otherwise we'd be setting the value from this devices' `deviceState` entry as the global lock state, which the next device could pick up and use, e.g.

```console
$ synse server read rack-1 vec da7fbdfc8e962922685af9d0fac53379
INFO      TYPE      VALUE     UNIT      TIMESTAMP
          state     locked              Fri Mar 22 18:14:58 UTC 2019
$ synse server read rack-1 vec 7218791ef7e9f9c0a85a150afaf6db3a
INFO      TYPE      VALUE     UNIT      TIMESTAMP
          state     locked              Fri Mar 22 18:15:02 UTC 2019
$ synse server write rack-1 vec da7fbdfc8e962922685af9d0fac53379 unlock
TRANSACTION ID         ACTION    DATA
biaiccietc9qaqkk7h40   unlock    
$ synse server read rack-1 vec da7fbdfc8e962922685af9d0fac53379
INFO      TYPE      VALUE                   UNIT      TIMESTAMP
          state     unlocked_electrically             Fri Mar 22 18:15:19 UTC 2019
$ synse server read rack-1 vec 7218791ef7e9f9c0a85a150afaf6db3a
INFO      TYPE      VALUE                   UNIT      TIMESTAMP
          state     unlocked_electrically             Fri Mar 22 18:15:23 UTC 2019
```

Reply:
note: I think I missed this same distinction in the LED PR /:

Reply:
All good, that actually cleans this up quite a bit. the dState explicit assignment was kind of bothering me a bit because it didn't make a ton of sense to set a value without it being tied directly to a device id. :+1: will clean up.